docs/source/user/getting_started/custom_task_schemas_and_workflows.rst

```diff
@@ -0,0 +1,357 @@
+Writing custom workflows
+########################
+
+MatFlow has a number of built-in :ref:`workflows <def_workflow>`, which use combinations of the
+built-in :ref:`task schemas <def_task_schema>`. 
+It is quite possible to mix and match these task schema into new workflows,
+and indeed to write your own task schemas to achieve a particular task.
+
+
+Workflow files
+---------------
+In-built matflow workflows are split up over a few different files,
+but for development, your workflow code can all go in one yaml file.
+The workflow template has a top-level key ``template_components``
+underneath which come the ``task_schema``, ``environments`` and ``command_files`` keys.
+
+The workflow itself goes under a different top-level `tasks` key.
+
+Components of a task schema
+----------------------------
+
+Required keys
+*****************
+- ``objective`` (this is a name or label for the schema)
+- ``actions`` (what the task schema actually "does")
+
+Optional keys
+*****************
+- ``inputs``
+- ``outputs``
+
+Matflow syntax
+--------------
+If you want to reference parameters in the action of your task schema,
+it should be done using this syntax:
+``<<parameter:your_parameter_name>>``.
+
+Similarly, commands defined in an environment can be used like this:
+``<<executable:your_executable>>``, and files defined as :ref:`command_files <def_command_files>`
+are referenced using ``<<file:your_command_file>>`` e.g.
+
+.. code-block:: console
+
+    actions:
+    - commands:
+      - command: <<executable:abaqus>> job=sub_script_check input=<<file:new_inp_file>> interactive
+
+
+Note that while command files can be referenced in an action, they cannot be referenced in this way as an input to a task schema.
+
+Python scripts however are executed slightly differently, and run the first
+function defined in your python file, which must have the same name as the python file.
```

Comment:
Not the first function. They run the function whose name is the same as the file name.

Reply:
Fixed